pkg/storage/pv.go

```diff
@@ -60,6 +62,76 @@ func (builder *PVBuilder) Exists() bool {
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
+// Delete removes a PersistentVolume from the apiClient if it exists.
+func (builder *PVBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the PersistentVolume %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		builder.Object = nil
```

Comment:
Please add log message above:
		glog.V(100).Infof("pv %s cannot be deleted because it does not exist", builder.Definition.Name)